### PR TITLE
Add a TypeCoercer for InputPaths

### PR DESCRIPTION
**Pull request by shs96c**

### CODE REVIEW

**Comment by asp2insp** [comment]
Can you write up a quick description please? I'm assuming this is related to the `query_paths` implementation?

**Comment by shs96c** [comment]
`TypeCoercer`s are used to convert from the raw values in the `BUCK` files to the strongly typed goodies in the `Description` "constructor arg types". This allows us to avoid the constant decanting from `SourcePath` to `InputPath` and back to `SourcePath` that we see in the `Zip` rule.

None of the other `TypeCoercer` implementations have docs as they're not user-facing classes

This change allows a `ModernBuildRule` to have a constructor arg like:

```
@BuckStyleImmutable
@Value.Immutable
interface AbstractFooArg extends CommonDescriptionArg {
  InputPath getChrome();
  @Value.NaturalOrder ImmutableSortedSet<InputPath> getComponents();
}
```

File: test/com/facebook/buck/rules/coercer/InputPathTypeCoercerTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+import static org.junit.Assert.assertEquals;
+
+import com.facebook.buck.model.BuildTarget;
+import com.facebook.buck.model.UnflavoredBuildTarget;
+import com.facebook.buck.rules.CellPathResolver;
+import com.facebook.buck.rules.DefaultBuildTargetSourcePath;
+import com.facebook.buck.rules.PathSourcePath;
+import com.facebook.buck.rules.TestCellPathResolver;
+import com.facebook.buck.rules.modern.InputPath;
```
Comment (asp2insp) [inline test/com/facebook/buck/rules/coercer/InputPathTypeCoercerTest.java]: Tests are failing due to the missing dependency on `rules/modern` for this test.

**Comment by shs96c** [comment]
Fixed the failing compilation and reformatted the test.

**Pull request closed without merging**